src/fcmp_pp/fcmp_pp_rust/src/lib.rs

```diff
@@ -107,12 +107,18 @@ const unsafe fn slice_from_raw_parts_0able<'a, T>(p: *const T, len: usize) -> &'
     }
 }
 
+// TODO: deprecate in favor of ed25519_point_from_bytes2
 fn ed25519_point_from_bytes(ed25519_point: *const u8) -> EdwardsPoint {
     let mut ed25519_point = unsafe { core::slice::from_raw_parts(ed25519_point, 32) };
-    // TODO: Return an error here (instead of unwrapping)
     <Ed25519>::read_G(&mut ed25519_point).unwrap()
 }
 
+// TODO: replace all usage of ed25519_point_from_bytes with this
+fn ed25519_point_from_bytes2(ed25519_point: [u8; 32]) -> EdwardsPoint {
```

Comment:
What's the difference with this function?

Reply:
I'm thinking we'll want to use byte slices instead of raw pointers for all the types currently passed into `ed25519_point_from_bytes`, and so `ed25519_point_from_bytes` will just do what this function is doing instead

Reply:
> And if we changed the API to not take raw pointers, but also provide a (compile-time fixed) length, if the calling user screws it up, they screw it up

This is what I was thinking. I think at the very least, using fixed 32 byte slices for data that is always going to be a fixed 32 bytes is clearer code that's easier to manage versus using a raw pointer. There are various ways calling users can screw things up in C++, I think this adds a layer of defense by making the code clearer.

Reply:
I could consolidate all usage of `ed25519_point_from_bytes` into this function in this PR if you agree with that approach (and fix merge conflicts). That way we're sure we don't end up a "2" lingering forever like that wallet thing.

Reply:
Thoughts on the goal to replace all raw pointers with byte slices when referencing data of size known at compile time, perhaps for a future PR(s)?

Reply:
At the API level or inside the Rust impl code?

Reply:
Ah okay I see, so in the C API you're not actually passing a fixed size of 32 as a function parameter, just modifying the function signature on the Rust side. Yeah, I'm fine with that. The safety is basically the same IMO, but it does add extra information for the reader. I'd support that change.

> And then instead of taking a raw pointer here, use a `[u8; 32]` instead.

Not to be that guy, but wouldn't it be a `&[u8; 32]`? Rust actually passes arrays by value, unlike C which decays them into pointers.

Reply:
> Not to be that guy, but wouldn't it be a &[u8; 32]?

You're right. And also nevermind, apparently it's not safe: https://users.rust-lang.org/t/creating-fixed-size-slice-from-pointer-passed-from-c/52717/10

Reply:
Like you *could* define a C struct `struct Buf64 { unsigned char buf[64]; };` and it's Rust analogue, and pass the struct containing the array by value. Maybe that's what he means. The downside of that is that that probably entails a stack-to-stack copy at the ABI level. Also it's ugly as hell. AFAIK, having a `&[u8; 32]` parameter on the Rust side is fine as long as you assume the C side passes a pointer to an area of 32 bytes which is readable.

Reply:
But I guess we can continue this discussion elsewhere too..